### PR TITLE
Plotting metrics

### PR DESCRIPTION
**Pull request by Hebruwu**

### CODE REVIEW

**Review by simonhkswan — COMMENTED** [review]
Looking good we just need to make sure the font file is included in the built python wheel.

This can be done by adding

```text
[tool.setuptools.package-data] 
synthesized = [
    "fonts/inter-v3-latin-regular.ttf",
]
```
To the pyproject.toml file

**Review by simonhkswan — APPROVED** [review]
Nice work!

File: src/synthesized_insight/metrics/metrics_usage.py [hunk 1]
```diff
@@ -1,68 +1,80 @@
...
     """Computes the correlation between each pair of columns in the given dataframe
     and returns the result in a dataframe"""
 
     def __init__(self, metric: TwoColumnMetric):
-        self.metric = metric
+        self._metric = metric
         self.name = f'{metric.name}_matrix'
 
-    def __call__(self, df: pd.DataFrame) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
+    def __call__(self, df: pd.DataFrame) -> Union[pd.DataFrame]:
```
Comment (hdaly0) [inline src/synthesized_insight/metrics/metrics_usage.py:35]: Don't think Union is needed here? Union is like "or" so here it's effectively saying "pd.DataFrame or"... with no 2nd option.

File: tests/test_plotting.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    fig_a = continuous_distribution_plot([df_half_a['fnlwgt'], df_half_b['fnlwgt']])
+    assert fig_a is not None
+
+    fig_b = continuous_distribution_plot([df_half_a['fnlwgt']])
+    assert fig_b is not None
+
+    assert fig_a != fig_b
+
+
+def test_plot_dataset(df_half_a, df_half_b):
```
Comment (hdaly0) [inline tests/test_plotting.py:97]: I think tests need more descriptive names. What is this actually testing? I'd call this one something like `test_plot_dataset_returns_non_None_figure_and_plots_new_figure_every_call` or similar

File: src/synthesized_insight/metrics/metrics_usage.py (hunk 1 above)
Comment (hdaly0) [reply src/synthesized_insight/metrics/metrics_usage.py:35]: Unless it's the expected return type of this call somewhere else...

**Pull request merged**